scipy/optimize/optimize.py

```diff
@@ -475,6 +475,36 @@ def function_wrapper(x, *wrapper_args):
     return ncalls, function_wrapper
 
 
+class _MaxFuncCallError(RuntimeError):
+    pass
+
+
+def _wrap_scalar_function_with_validation(function, args, maxfun):
+    # wraps a minimizer function to count number of evaluations
+    # and to easily provide an args kwd.
+    ncalls = [0]
```

Comment:
Why is this a list?

Reply:
This is for incrementing the function call numbers in the closure function. 
The original wrapper function is using same way:
https://github.com/scipy/scipy/blob/be2027dcf11745cc28c95f088436400471d520ae/scipy/optimize/optimize.py#L454-L462

Reply:
Oh right to have an object and have the changes in other scopes.